### DISCUSSION TITLE
Rich text editor / Editor with image support / Copypaste image to an editor

### DISCUSSION THREAD

**Discussion by open-domain** [eclipse-langium/langium#889] [Ideas]
Hello Team

Thanks a lot for this amazing framework, You guys are doing amazing job. I am enjoing and having fun with it.

Does Langium supports Rich text editors? AFIK, atleast I see no support in grammer. My expectation/idea is simple(actually not), i want to add a flowchart or maths formula or some tables in picture(jpeg, png) format to the editor(either by some way of insert option or by copy-paste from clipboard). I know vscode supports image rendering, but woundn't this be cool if langium has this support(for eg, take the image and generate back to output through generator or cross reference from this image to another element, etc) ?

**Reference:** 
https://docs.devexpress.com/CodeRushForRoslyn/120417/visualization-tools/rich-comments
https://github.com/lukesdm/image-comments
https://marketplace.visualstudio.com/items?itemName=MariusBancila.memefulcomments

MfG
Maha

**Comment by msujew** [comment]
Hey @open-domain,

rich text editors are quite out of scope for Langium. Aside from being technically complicated, the language server protocol (on which Langium builds all of its language services) offers no support for it. While an additional protocol can be built on top of JSON-RPC to enable such features, no language client exists that is capable of handling this protocol.

You can in theory create your own frontend/language client for this and use Langium as a backing infrastructure for serialization etc., but the framework is intended for code editor use only and everything else beyond what the LSP supports will likely increase maintenance costs massively.

**Reply by open-domain** [reply]
Thanks a lot for your answers.
